Source/VMware.vSphereDSC/VMware.vSphereDSC.build.ps1

```diff
@@ -0,0 +1,101 @@
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+$script:ModuleRoot = $PSScriptRoot
+$script:ProjectRoot = (Get-Item -Path $script:ModuleRoot).Parent.Parent.FullName
+$script:ModuleName = Split-Path -Path $script:ModuleRoot -Leaf
+
+$script:PsmPath = Join-Path -Path $script:ModuleRoot -ChildPath "$($script:ModuleName).psm1"
+$script:PsdPath = Join-Path -Path $script:ModuleRoot -ChildPath "$($script:ModuleName).psd1"
+
+$script:LicensePath = Join-Path -Path $script:ProjectRoot -ChildPath "LICENSE.txt"
+# This is used to skip the lines from LICENSE.txt containing the repository name and the empty line after it.
+$script:LicenseSkipLines = 2
+$script:LicenseFileContent = Get-Content -Path $script:LicensePath | Select-Object -Skip $script:LicenseSkipLines
+
+$script:ImportFolders = @('Enums', 'Classes', 'DSCResources')
+$script:DSCResourcesFolder = Join-Path -Path $script:ModuleRoot -ChildPath "DSCResources"
+
+# Add License to psm1 file.
+"<#" | Out-File -FilePath $script:PsmPath -Encoding Default
+$script:LicenseFileContent | ForEach-Object { $_ | Out-File -FilePath $script:PsmPath -Encoding Default -Append }
+"#>" + [System.Environment]::NewLine | Out-File -FilePath $script:PsmPath -Encoding Default -Append
+
+# Add helper module to psm1 file.
+"Using module '.\VMware.vSphereDSC.Helper.psm1'" | Out-File -FilePath $script:PsmPath -Encoding Default -Append
+
+# Updating VMware.vSphereDSC.psm1 content with enums, classes and DSC Resources.
+foreach ($folder in $script:ImportFolders) {
+    $currentFolder = Join-Path -Path $script:ModuleRoot -ChildPath $folder
+
+    if (Test-Path -Path $currentFolder) {
+        $files = Get-ChildItem -Path $currentFolder -File -Filter '*.ps1'
+
+        foreach ($file in $files) {
+            $index = 1
+            $startLine = 0
+            $endLine = 0
+
+            $fileContent = Get-Content -Path $file.FullName
+            $fileContent | ForEach-Object {
+                if ($_ -Like '*Copyright*') {
+                    $startLine = $index
+                }
+
+                if ($_.Trim().EndsWith('#>') -and $startLine -ne 0) {
+                    $endLine = $index
```

Comment:
break the loop at this point instead of playing with startindex. Have in mind Foreach-Object is a cmdleta and break keyword won't work for it.